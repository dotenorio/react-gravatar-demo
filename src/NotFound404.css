a {
  color: #444;
}

a:hover {
  opacity: 0.6;
}

p {
  margin-bottom: 20px;
}

p:last-child {
  margin-bottom: 0;
}

@media only screen and (max-width: 768px) {
  p {
    text-align: center;
  }
}