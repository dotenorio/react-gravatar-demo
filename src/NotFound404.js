import React from 'react'
import Layout from "./Layout";

function NotFound404 () {
  return (
    <div>
      <Layout title="Oops!!">
        <p>Page Not Found</p>
        <br />
        <p>:(</p>
      </Layout>
    </div>
  );
}

export default NotFound404