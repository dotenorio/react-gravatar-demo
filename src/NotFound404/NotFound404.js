import React from 'react'
import { NavLink } from "react-router-dom";

import Layout from "../Layout/Layout";
import './NotFound404.css';

function NotFound404 () {
  return (
    <div>
      <Layout title="Oops!!">
        <p>Page Not Found</p>
        <p>:(</p>
        <p>
          <NavLink to="/" exact activeClassName="active" title="back to Home Page">back</NavLink>
        </p>
      </Layout>
    </div>
  );
}

export default NotFound404