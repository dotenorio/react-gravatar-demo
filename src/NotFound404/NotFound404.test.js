import React from 'react';
import { MemoryRouter } from 'react-router-dom'
import renderer from 'react-test-renderer';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() });

import NotFound404 from './NotFound404';

test('Load NotFound404', () => {
  const wrapper = shallow(
    <NotFound404 />
  );

  expect(wrapper.contains(<p>Page Not Found</p>)).toEqual(true);
  expect(wrapper.contains(<p>:(</p>)).toEqual(true);
})

test('NotFound404 renders correctly', () => {
  const tree = renderer
    .create(<MemoryRouter><NotFound404 /></MemoryRouter>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});