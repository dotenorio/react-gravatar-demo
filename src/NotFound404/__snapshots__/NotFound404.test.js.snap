// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFound404 renders correctly 1`] = `
<div>
  <div>
    <div
      className="header"
    >
      <h1>
        Oops!!
      </h1>
      <ul>
        <li>
          <a
            aria-current="page"
            className="active"
            href="/"
            onClick={[Function]}
            style={Object {}}
            title="Home Page"
          >
            Home
          </a>
        </li>
        <li>
          <a
            aria-current={null}
            href="/about"
            onClick={[Function]}
            title="About Project"
          >
            About
          </a>
        </li>
      </ul>
    </div>
    <div
      className="content"
    >
      <p>
        Page Not Found
      </p>
      <p>
        :(
      </p>
      <p>
        <a
          aria-current="page"
          className="active"
          href="/"
          onClick={[Function]}
          style={Object {}}
          title="back to Home Page"
        >
          back
        </a>
      </p>
    </div>
  </div>
</div>
`;
